src/main/java/com/suse/salt/netapi/event/AbstractEventStream.java

```diff
@@ -66,10 +61,8 @@ public int getListenerCount() {
      * @param event the event
      */
     protected void notifyListeners(Event event) {
-        synchronized (listeners) {
-            for (EventListener listener : new ArrayList<>(listeners)) {
-                listener.notify(event);
-            }
+        for (EventListener listener : new ArrayList<>(listeners)) {
```

Comment:
if i remember correctly the new `ArrayList` here was done in case the callback of a listener modifies the listener list resulting in a `ConcurentModificationException`. From the `CopyOnWriteArray` docs it seems that it does not have this problem. 

> ... and the iterator isguaranteed not to throw {@code ConcurrentModificationException}

Reply:
Right, fixing.